### PR TITLE
remove javax.json and replace with FasterXML

### LINKED ISSUE #11544: update javax.json (1.1.6) to jakarta.json (2.0.1)
### Description

update javax.json (1.1.6) to jakarta.json (2.0.1)

This package appears to only be in used in
https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java

### Discussion

_No response_

### Motivation

I'm running into more and more projects that are using the newer library.

### Details

_No response_

### PR DESCRIPTION
**Pull request by ebremer**
remove javax.json and replace with FasterXML to be consistent with the rest of the project.

Closes #11544

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java [hunk 1]
```diff
@@ -159,13 +159,18 @@ private Map<String, String> supportedLocales(Properties messages) {
     
     private String messagesToJsonString(Properties props) {
         if (props == null) return "";
-        
-        JsonObjectBuilder json = Json.createObjectBuilder();
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (stianst) [inline services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java:162]: Should be possible to replace with:

JsonSerialization.writeValueAsString(props)
```suggestion
        ObjectMapper mapper = new ObjectMapper();        
```
Using [JsonSerialization](https://github.com/keycloak/keycloak/blob/8ea8aaebcf3ec7166607f79141f40ba7a2eecfaa/core/src/main/java/org/keycloak/util/JsonSerialization.java#L38) uses a shared ObjectMapper + makes it possible to configure globally for the server if pretty-printing is used or not.

**Review by stianst — COMMENTED** [review]
Should also remove javax.json from POMs now that it is no longer used.

**Comment by stianst** [comment]
Thanks, will be good to get this sorted out. Added some comments on improvements to this PR.

**Comment by stelewis-redhat** [comment]
I put together a PR that included these changes, but noticed a difference in the account console in the JSON string output:

GOOD: "setUpNew":"Set up {{param_0}}",
BAD: "setUpNew":"Set up {0}",

I will investigate. The symptom is that on the SigningInPage I see a message for "Set up {0}" and "{0} is not set up." instead of the correct messages.

File: services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java (hunk 1 above)
Comment (stelewis-redhat) [reply services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java:162]: @ebremer This is looking good. I tried creating a PR and chose the wrong json javax lib to remove. I think the only outstanding thing here is probably using JsonSerialization instead of ObjectMapper directly.

File: services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java [hunk 2]
```diff
@@ -159,13 +158,11 @@ private Map<String, String> supportedLocales(Properties messages) {
     
     private String messagesToJsonString(Properties props) {
         if (props == null) return "";
-        
-        JsonObjectBuilder json = Json.createObjectBuilder();
-        for (String prop : props.stringPropertyNames()) {
-            json.add(prop, convertPropValue(props.getProperty(prop)));
+        try {
+            return JsonSerialization.writeValueAsString(props);
```
[the remark this answers is not in the document]
Comment (ebremer) [reply services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java:167]: @stelewis-redhat it does.  I added the loop back in.

**Comment by stelewis-redhat** [comment]
@abstractj @stianst I think this is more ready now. @ebremer

File: services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java [hunk 3]
```diff
@@ -71,7 +70,8 @@ public AccountConsole(RealmModel realm, ClientModel client, Theme theme) {
         this.realm = realm;
         this.client = client;
         this.theme = theme;
-        this.authManager = new AppAuthManager();
+        //this.authManager = new AppAuthManager();
+        this.authManager = null;
```
Comment (stianst) [inline services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java:74]: Was this a mistake? Seems irrelevant and a breaking change I think.

**Review by stianst — COMMENTED** [review]
LGTM now except the changes to the constructor, and needs a rebase :/

**Comment by abstractj** [comment]
@ebremer could you please rebase and review Stian's comments?

File: services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java (hunk 3 above)
Comment (ebremer) [reply services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java]: Temporary code I was using but that I forgot to remove before my commit

**Review by stianst — CHANGES_REQUESTED** [review]
Looks like something went wrong here when rebasing, as it looks like it's now reverting quite a few changes that where made to AccountConsole.

**Comment by stelewis-redhat** [comment]
> @stianst I think it looks correct now. The removal of the javax.json library from the main pom is there now too. It had been dropped too.

@ebremer I'm seeing:

"Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1-jboss-1:compile (default-compile) on project keycloak-services: Compilation failure: Compilation failure: 
Error:  /home/runner/work/keycloak/keycloak/services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java:[15,18] package javax.json does not exist
Error:  /home/runner/work/keycloak/keycloak/services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java:[16,18] package javax.json does not exist
Error:  -> [Help 1]"

In the build results (that first one listed that fails in 2 minutes): https://github.com/keycloak/keycloak/runs/7046755910?check_suite_focus=true

**Comment by stelewis-redhat** [comment]
> 

I did a local build and ran it and things seem to look good there in the account console.

**Comment by stelewis-redhat** [comment]
@thomasdarimont Do you have bandwidth to look at this. Between @ebremer and myself I think we're good, and we don't have any conflicts anymore. I got a little too eager last time thinking we were ready before Erich fixed the conflicts. Thanks in advance.

**Review by stianst — APPROVED** [review]

**Comment by stelewis-redhat** [comment]
Congratulations @ebremer !!
